File: VonexDealer.API/Models/UB/RatePlan.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace VonexDealer.API.Models.UB
{
    public class RatePlan
    {
        public int ID { get; set; }

        public int? RatePlanID { get; set; }

        public string RatePlanDescription { get; set; }

        public string GroupName { get; set; }

        public string UsageType { get; set; }

        public bool? Handset { get; set; }

        public decimal? MonthlyRetail { get; set; }

        public Single? MonthlyProfitPercent { get; set; }

        public string RatePlanPDF { get; set; }

        public int? DataAllowance { get; set; }

        public bool? MonitorDataUsage { get; set; }

        public int? ContractMonths { get; set; }

        public bool? PayInitialComms { get; set; }

        public bool? HasSpecialOverride { get; set; }

        public bool DealerSupplied { get; set; }
        public bool MixnMatch { get; set; }
        public bool AdditionalServices { get; set; }
        public bool Residential { get; set; }

    }



}

File: VonexDealer.Vue/ClientApp/models/ratePlan.js
export class ratePlan {
    constructor({
iD = null,
        ratePlanID = null,
        ratePlanDescription = null,
        groupName = null,
        usageType = null,
        handset = null,
        monthlyRetail = null,
        monthlyProfitPercent = null,
        ratePlanPDF = null,
        dataAllowance = null,
        contractMonths = null,
        payInitialComms = null,
        hasSpecialOverride = null,
        dealerSupplied = null,
        mixnMatch = null,
        additionalServices = null,
        residential = null,
    } = {}) {
        this.iD = iD;
        this.ratePlanID = ratePlanID;
        this.ratePlanDescription = ratePlanDescription;
        this.groupName = groupName;
        this.usageType = usageType;
        this.handset = handset;
        this.monthlyRetail = monthlyRetail;
        this.monthlyProfitPercent = monthlyProfitPercent;
        this.ratePlanPDF = ratePlanPDF;
        this.dataAllowance = dataAllowance;
        this.contractMonths = contractMonths;
        this.payInitialComms = payInitialComms;
        this.hasSpecialOverride = hasSpecialOverride;
        this.additionalServices = additionalServices;
        this.residential = residential;
        this.dealerSupplied = dealerSupplied;
        this.mixnMatch = mixnMatch;
    }
}

export function createRatePlan(data) {
    return Object.freeze(new ratePlan(data));
}
